Hi Aaron, from unit tests results below, that you can also see in https://github.com/alien4cloud/alien4cloud-go-client/actions there is a failure because GetTopologies() is still referenced in tests:
```
alien4cloud/topology_test.go:68:29: topoService.GetTopologies undefined (type *topologyService has no field or method GetTopologies)
```
But it should be better to not change ` GetTopologies()` because it is actually used by another team.

For the new function `GetTopologyByID()` 
Actually there is a function that should be doing what you need in topology service,
at https://github.com/alien4cloud/alien4cloud-go-client/blob/master/alien4cloud/topology.go#L35
```golang
	// Returns Topology details for a given application and environment
	GetTopology(ctx context.Context, appID string, envID string) (*Topology, error)
```
This is not exactly your `GetTopologyByID()` as you are passing a topologyID.
And the existing `GetTopology()` expects an appID and an envID.
This function `GetTopology()` is first calling `GetTopologyID()` to get a topologyID from appID and an envID.
Then executes exactly the same code as your new code to get the topology.

What do you think, do your prefer to still have `GetTopologyByID()` if your code has a toplogyID but not necesseraly an appID and envID in memory at the thime you call    `GetTopologyByID()` ?
In this case no problem, you could just refactor `GetTopology()` to call your new `GetTopologyByID()` once it has a topolgyID, just to avoid code duplication.

Otherwise, ok for new fields in structs!

--- a/alien4cloud/topology.go
+++ b/alien4cloud/topology.go
@@ -58,7 +58,7 @@ type TopologyService interface {
 	// Deletes a policy from the topology
 	DeletePolicy(ctx context.Context, a4cCtx *TopologyEditorContext, policyName string) error
 	// Returns a list of topologyIDs available topologies
-	GetTopologyIDs(ctx context.Context, query string) ([]BasicTopologyInfo, error)
+	GetTopologies(ctx context.Context, query string) ([]BasicTopologyInfo, error)
 	// Returns Topology details for a given TopologyID
 	GetTopologyByID(ctx context.Context, a4cTopologyID string) (*Topology, error)
 }
@@ -193,24 +193,13 @@ func (t *topologyService) GetTopology(ctx context.Context, appID string, envID s
 	a4cTopologyID, err := t.GetTopologyID(ctx, appID, envID)
 
 	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get A4C application topology ID for app %s and env %s", appID, envID)
+	}
+
+	res, err := t.GetTopologyByID(ctx, a4cTopologyID)
+
+	if err != nil {
 		return nil, errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s", appID, envID)
-	}
-
-	response, err := t.client.doWithContext(ctx,
-		"GET",
-		fmt.Sprintf("%s/topologies/%s", a4CRestAPIPrefix, a4cTopologyID),
-		nil,
-		[]Header{contentTypeAppJSONHeader},
-	)
-
-	if err != nil {
-		return nil, errors.Wrapf(err, "Cannot get the topology content for application '%s' in '%s' environment", appID, envID)
-	}
-
-	res := new(Topology)
-	err = processA4CResponse(response, res, http.StatusOK)
-	if err != nil {
-		return nil, errors.Wrapf(err, "Cannot convert the body of topology get data for application '%s' in '%s' environment", appID, envID)
 	}
 
 	return res, nil
@@ -523,7 +512,7 @@ func (t *topologyService) SaveA4CTopology(ctx context.Context, a4cCtx *TopologyE
 	return processA4CResponse(response, nil, http.StatusOK)
 }
 
-func (t *topologyService) GetTopologyIDs(ctx context.Context, query string) ([]BasicTopologyInfo, error) {
+func (t *topologyService) GetTopologies(ctx context.Context, query string) ([]BasicTopologyInfo, error) {
 
 	getTopoJSON, err := json.Marshal(
 		searchRequest{

